### ISSUE TITLE
"cypress run" fails but "cypress open" works

### ISSUE THREAD

**Issue by darthmeme** [cypress-io/cypress#6697]
### Current behavior:
The issue here is with trying to get a label that contains some specific text.
`cy.get('label').contains('Select 2')`

When running `cypress open` locally and running my e2e tests, everything passes.

On the other hand, when running the same specs in CircleCI, it fails. I can reproduce this locally when running `cypress run`

### Desired behavior:
`cypress run` should be passing

### Test code to reproduce
https://github.com/darthmeme/cypress-test

In this example, I'm using Chrome.
After installing dependencies:
- Start the app locally with `yarn serve`
- In another terminal, run `yarn cy:open` (uses `cypress open`) and choose Chrome. Run all specs and they should all pass

Issue occurs here:
- Run `yarn cy:run` (uses `cypress run` and automatically uses Chrome), and the specs will fail

Any insights or help will be greatly appreciated!

### Versions
Cypress: `3.8.3` (also reproduced on `4.1.0`
OS: `macOS Catalina`
Browser: `Chrome`

**Comment by jennifer-shehane** [comment]
Thank you for providing a reproducible example! I can recreate this. I went back to Cypress 3.4.1 to see if this still reproduces - it's still the same error there. 

Seems to not be finding the labels during `cypress run`.

`cypress open` in Chrome

<img width="1009" alt="Screen Shot 2020-03-11 at 3 33 54 PM" src="https://user-images.githubusercontent.com/1271364/76399977-f3437d80-63ad-11ea-80bf-e636731260c0.png">

`cypress run` in Chrome

<img width="890" alt="Screen Shot 2020-03-11 at 3 34 24 PM" src="https://user-images.githubusercontent.com/1271364/76400032-08b8a780-63ae-11ea-9746-75a7cf7b025b.png">

I was able to get this to work by using `cy.contains('label', 'Select 2')` instead of `cy.get('label').contains('Select 2')`. 

I think this may be a duplicate of this issue: https://github.com/cypress-io/cypress/issues/2640 and there just happens to be some variability in the runtime between `cypress open` and `cypress run` - where perhaps the `label` is rerendering, so is gone by the time the `.contains()` runs.
